TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/snapshot.py

```diff
@@ -309,14 +267,120 @@ class SnapshotNotStarted(Exception):
     pass
 
 
-class SnapshotStatusUnavaliable(Exception):
+class SnapshotStatusUnavailable(Exception):
     pass
 
 
+class SnapshotStatus(BaseModel):
+    status: str
+    percentage_completed: float
+    eta_ms: float | None
+    started: datetime | None = None
+    finished: datetime | None = None
+    data_total_bytes: int | None
+    data_processed_bytes: int | None
+    data_throughput_bytes_avg_sec: float | None
+    shard_total: int | None
+    shard_complete: int | None
+    shard_failed: int | None
+
+    class Config:
+        from_attributes = True
```

Comment:
Looks like you could potentially remove this inner class with: `model_config = ConfigDict(from_attributes=True)`